From d93f4ea7e3eab50573a736a02e17bde0de921bb1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 22:26:21 +0800
Subject: [PATCH] Revert "USB: gadget: f_rndis: Add platform data for RNDIS
 vendor ID and MAC address."

This reverts commit 7dc693330c1fda407643b95c876fc072845c0593.
---
 drivers/usb/gadget/f_rndis.c          | 46 ++++++---------------------
 include/linux/usb/android_composite.h |  8 -----
 2 files changed, 10 insertions(+), 44 deletions(-)

diff --git a/drivers/usb/gadget/f_rndis.c b/drivers/usb/gadget/f_rndis.c
index 7c4fb4d022ab..21bf50f5bbb8 100644
--- a/drivers/usb/gadget/f_rndis.c
+++ b/drivers/usb/gadget/f_rndis.c
@@ -23,7 +23,7 @@
 /* #define VERBOSE_DEBUG */
 
 #include <linux/kernel.h>
-#include <linux/platform_device.h>
+#include <linux/device.h>
 #include <linux/etherdevice.h>
 #include <linux/usb/android_composite.h>
 
@@ -292,10 +292,6 @@ static struct usb_gadget_strings *rndis_strings[] = {
 	NULL,
 };
 
-#ifdef CONFIG_USB_ANDROID_RNDIS
-static struct usb_ether_platform_data *rndis_pdata;
-#endif
-
 /*-------------------------------------------------------------------------*/
 
 static struct sk_buff *rndis_add_header(struct gether *port,
@@ -698,12 +694,11 @@ rndis_bind(struct usb_configuration *c, struct usb_function *f)
 	rndis_set_param_medium(rndis->config, NDIS_MEDIUM_802_3, 0);
 	rndis_set_host_mac(rndis->config, rndis->ethaddr);
 
-#ifdef CONFIG_USB_ANDROID_RNDIS
-	if (rndis_pdata) {
-		if (rndis_set_param_vendor(rndis->config, rndis_pdata->vendorID,
-					rndis_pdata->vendorDescr))
-			goto fail;
-	}
+#if 0
+// FIXME
+	if (rndis_set_param_vendor(rndis->config, vendorID,
+				manufacturer))
+		goto fail0;
 #endif
 
 	/* NOTE:  all that is done without knowing or caring about
@@ -855,35 +850,15 @@ int __init rndis_bind_config(struct usb_configuration *c, u8 ethaddr[ETH_ALEN])
 #ifdef CONFIG_USB_ANDROID_RNDIS
 #include "rndis.c"
 
-static int __init rndis_probe(struct platform_device *pdev)
-{
-	rndis_pdata = pdev->dev.platform_data;
-	return 0;
-}
+// FIXME - using bogus MAC address for now
 
-static struct platform_driver rndis_platform_driver = {
-	.driver = { .name = "rndis", },
-	.probe = rndis_probe,
-};
+static u8 ethaddr[ETH_ALEN] = { 11, 22, 33, 44, 55, 66 };
 
 int rndis_function_bind_config(struct usb_configuration *c)
 {
-	int ret;
-
-	if (!rndis_pdata) {
-		printk(KERN_ERR "rndis_pdata null in rndis_function_bind_config\n");
-		return -1;
-	}
-
-	printk(KERN_INFO
-		"rndis_function_bind_config MAC: %02X:%02X:%02X:%02X:%02X:%02X\n",
-		rndis_pdata->ethaddr[0], rndis_pdata->ethaddr[1],
-		rndis_pdata->ethaddr[2], rndis_pdata->ethaddr[3],
-		rndis_pdata->ethaddr[4], rndis_pdata->ethaddr[5]);
-
-	ret = gether_setup(c->cdev->gadget, rndis_pdata->ethaddr);
+	int ret = gether_setup(c->cdev->gadget, ethaddr);
 	if (ret == 0)
-		ret = rndis_bind_config(c, rndis_pdata->ethaddr);
+		ret = rndis_bind_config(c, ethaddr);
 	return ret;
 }
 
@@ -895,7 +870,6 @@ static struct android_usb_function rndis_function = {
 static int __init init(void)
 {
 	printk(KERN_INFO "f_rndis init\n");
-	platform_driver_register(&rndis_platform_driver);
 	android_register_function(&rndis_function);
 	return 0;
 }
diff --git a/include/linux/usb/android_composite.h b/include/linux/usb/android_composite.h
index 50889ba138b9..328016813dd5 100644
--- a/include/linux/usb/android_composite.h
+++ b/include/linux/usb/android_composite.h
@@ -18,7 +18,6 @@
 #define	__LINUX_USB_ANDROID_H
 
 #include <linux/usb/composite.h>
-#include <linux/if_ether.h>
 
 struct android_usb_function {
 	struct list_head	list;
@@ -81,13 +80,6 @@ struct usb_mass_storage_platform_data {
 	int nluns;
 };
 
-/* Platform data for USB ethernet driver. */
-struct usb_ether_platform_data {
-	u8	ethaddr[ETH_ALEN];
-	u32	vendorID;
-	const char *vendorDescr;
-};
-
 extern void android_usb_set_connected(int on);
 
 extern void android_register_function(struct android_usb_function *f);
-- 
2.35.3

